samples-generator/lib/create3dtilesBatchTableExt.js

```diff
@@ -0,0 +1,137 @@
+var defined = Cesium.defined;
+var Extensions = require ('./Extensions');
+var batchTableExtensionName = 'CESIUM_3dtiles_batch_table';
+
+function sortByByteOffset(a, b) {
+    if (a.byteOffset < b.byteOffset ){
+        return -1;
+    }
+
+    if (a.byteOffset > b.byteOffset) {
+        return 1;
+    }
+
+    return 0;
+}
+
+function assertHumanReadableBatchTableValueArraysHaveIdenticalLength(humanBatchTables) {
+    if (humanBatchTables.length === 0) {
+        return 0;
+    }
+
+    var firstLength = humanBatchTables[0].value.length;
+    for (var i = 1; i < humanBatchTables.length; ++i) {
+        if (humanBatchTables[i].value.length !== firstLength) {
+            var badBatchTable = humanBatchTables[i].name;
+            var badLength = humanBatchTables[i].value.length;
+            throw new Error(badBatchTable + ' has incorrect length of: '  + badLength + ', should match the first length: ' + firstLength);
+        }
+    }
+
+    return firstLength;
+}
+
+/**
+ * @typedef humanReadableBatchTableValue
+ * @type {Array} values An array of arbitrary values in human readable form (e.g ["A", "B", "C"] or [1.3, 4.3, 1.5])
+ */
+
+/**
+ * @typedef binaryReadableBatchTableValue
+ * @type {String} name Name of the batchTableAttribute (e.g to be placed into the accessor)
+ * @type {Number} byteoffset ByteOffset of the batchTableAttribute
+ * @type {Number} byteLength Length of the attribute in bytes
+ * @type {Number} count Count of logical number of elements in the attribute (eg 9 floating point numbers with a vec3 type means count = 3)
+ * @type {Number} componentType WebGL enum of the component type (e.g GL_UNSIGNED_BYTE / 0x1401)
+ */
+
+/**
+ * @typedef batchTableAttribute
+ * @type {Object.<string, humanReadableBatchTableValue|binaryReadableBatchTableValue>} batchTableAttribute;
+ */
+
+/**
+ * Iterates over batchTableAttributes and adorns the provided gltf object with them via
+ * the CESIUM_3dtiles_batch_table extension
+ *
+ * @param {Object} gltf The gltf object to edit
+ * @param {batchTableAttribute} batchTableAttributes List of batch table attributes. The function will intelligently try to detect
+ *                                                   if the attribute is a human readable value (a key value where the value is
+ *                                                   just an array of non-binary data) or if the values are more keys describing the
+ *                                                   layout of binary data in a binary buffer.
+ * @param {Buffer}              sharedBinaryBuffer   Binary buffer. Must be defined if using at least 1 binary batch attribute.
+ *                                                   This function currently assumes that all of the binary batch attributes in
+ *                                                   batchTableAttributes are directly referring to this buffer.
+ */
+function create3dtilesBatchTableExt(gltf, batchTableAttributes, sharedBinaryBuffer) {
```

Comment:
Did you see this comment: https://github.com/CesiumGS/3d-tiles-validator/pull/186/files#r386654358?